app/controllers/manage_proposal_controller.rb

```diff
@@ -70,4 +91,41 @@ def user_has_approver_permissions?
   def unauthorized?(response)
     response.status == 401
   end
+
+  def publish_delete_proposal(proposal, provider)
+    search_response = cmr_client.get_collections({ 'native_id': proposal['native_id'], 'provider_id': provider }, token)
+
+    if search_response.body['hits'].to_s != '1'
+      # If the search has more than one hit or 0 hits, the record was not
+      # uniquely identified from it's native ID.
+      flash[:error] = I18n.t('controllers.manage_proposals.publish.flash.delete.not_found_error')
+      Rails.logger.info("Could not complete delete request from proposal with short name: #{proposal['short_name']} and id: #{proposal['id']} because it could not be located.")
+    elsif !search_response.body['items'][0]['meta']['granule-count'].zero?
+      # Do not allow the deletion of collections which have granules
+      flash[:error] = I18n.t('controllers.manage_proposals.publish.flash.delete.granules_error')
+      Rails.logger.info("Could not complete delete request from proposal with short name: #{proposal['short_name']} and id: #{proposal['id']} because it had granules at the time of the delete attempt.")
+    else
+      cmr_response = cmr_client.delete_collection(provider, proposal['native_id'], token)
+
+      if cmr_response.success?
+        flash[:success] = I18n.t('controllers.manage_proposals.publish.flash.delete.success')
+        Rails.logger.info("Audit Log: Collection with native_id #{proposal['native_id']} was deleted for #{provider} by #{session[:urs_uid]} by proposal with short name: #{proposal['short_name']} and id: #{proposal['id']}.")
+      else
+        flash[:error] = I18n.t('controllers.manage_proposals.publish.flash.delete.error')
+        Rails.logger.info("User: #{current_user.urs_uid} could not delete a collection with native_id #{proposal['native_id']} based on proposal with short name: #{proposal['short_name']} and id: #{proposal['id']}")
+      end
+    end
+  end
+
+  def publish_create_or_update_proposal(proposal, provider)
+    cmr_response = cmr_client.ingest_collection(proposal['draft'].to_json, provider, proposal['native_id'], token)
+
+    if cmr_response.success?
+      flash[:success] = I18n.t('controllers.manage_proposals.publish.flash.create.success')
+      Rails.logger.info("Audit Log: Draft #{proposal['entry_title']} was published by #{current_user.urs_uid} in provider: #{provider} by proposal with short name: #{proposal['short_name']} and id: #{proposal['id']}")
+    else
+      flash[:error] = I18n.t('controllers.manage_proposals.publish.flash.create.error')
+      Rails.logger.info("User: #{current_user.urs_uid} could not publish proposal with short name: #{proposal['short_name']} and id: #{proposal['id']} in the CMR. The CMR provided the following reasons: #{cmr_response.errors.inspect}")
```

Comment:
Copied from CollectionDraftsController, but can you separate out the errors part and add a line like `Rails.logger.error("Ingest Collection Metadata Error: #{ingested_response.clean_inspect}")`